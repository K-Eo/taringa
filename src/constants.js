export const REPLACE = 'REPLACE'
export const PUSH = 'PUSH'
