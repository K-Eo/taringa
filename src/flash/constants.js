export const ADD = 'flash/ADD'
