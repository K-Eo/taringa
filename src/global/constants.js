export const filters = {
  hot: {
    displayName: 'Destacados',
    exact: true,
    id: 'hot',
    pathname: '/global',
    url: '/feed/global',
  },
  recents: {
    displayName: 'Recientes',
    exact: false,
    id: 'recents',
    pathname: '/global/recents',
    url: '/feed/global?sort=recent',
  },
  tops: {
    displayName: 'Tops',
    exact: false,
    id: 'tops',
    pathname: '/global/tops',
    url: '/feed/global/tops/week',
  },
}
