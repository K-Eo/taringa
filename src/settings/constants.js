export const ITEM_BIG = 'settings/ITEM_BIG'
export const ITEM_MEDIUM = 'settings/ITEM_MEDIUM'
export const ITEM_SMALL = 'settings/ITEM_SMALL'

export const CHANGE_ITEM_SIZE = 'settings/CHANGE_ITEM_SIZE'
