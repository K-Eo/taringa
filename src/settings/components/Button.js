import React from 'react'
import classNames from 'classnames'

import { ITEM_MEDIUM } from '../constants'

const Button = ({ active, className, icon, size, onChangeSize, ...rest }) => {
  const classes = classNames(className, 'btn px-1 py-0 btn-light', {
    active: active === size,
  })

  return (
    <button {...rest} className={classes} onClick={onChangeSize}>
      <i className={icon} />
    </button>
  )
}

Button.defaultProps = {
  active: ITEM_MEDIUM,
  icon: 'ta-items-medium',
  size: ITEM_MEDIUM,
  onChangeSize: () => {},
}

export default Button
