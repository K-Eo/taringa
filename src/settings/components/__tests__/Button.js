import React from 'react'

import Button from '../Button'
import { ITEM_BIG, ITEM_MEDIUM } from '../../constants'

describe('Button', () => {
  it('exists', () => {
    expect(Button).toBeDefined()
  })

  it('renders', () => {
    const wrapper = shallow(<Button />)

    expect(wrapper).toMatchSnapshot()
  })

  it('sets icon', () => {
    const wrapper = shallow(<Button icon="fa fa-circle" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('sets size', () => {
    const wrapper = shallow(<Button size="foobar" active="foobar" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('sets active class', () => {
    const wrapper = shallow(<Button size={ITEM_BIG} />)

    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ size: ITEM_MEDIUM })

    expect(wrapper).toMatchSnapshot()
  })

  it('handles onChangeSize', () => {
    const mock = jest.fn()

    const wrapper = shallow(<Button onChangeSize={mock} />)

    wrapper.find('button').simulate('click')

    expect(mock).toHaveBeenCalled()
  })
})
