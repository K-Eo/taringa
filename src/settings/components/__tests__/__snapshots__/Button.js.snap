// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<button
  className="btn px-1 py-0 btn-light active"
  onClick={[Function]}
>
  <i
    className="ta-items-medium"
  />
</button>
`;

exports[`Button sets active class 1`] = `
<button
  className="btn px-1 py-0 btn-light"
  onClick={[Function]}
>
  <i
    className="ta-items-medium"
  />
</button>
`;

exports[`Button sets active class 2`] = `
<button
  className="btn px-1 py-0 btn-light active"
  onClick={[Function]}
>
  <i
    className="ta-items-medium"
  />
</button>
`;

exports[`Button sets icon 1`] = `
<button
  className="btn px-1 py-0 btn-light active"
  onClick={[Function]}
>
  <i
    className="fa fa-circle"
  />
</button>
`;

exports[`Button sets size 1`] = `
<button
  className="btn px-1 py-0 btn-light active"
  onClick={[Function]}
>
  <i
    className="ta-items-medium"
  />
</button>
`;
